applications/tari_base_node/src/commands/command/period_stats.rs

```diff
@@ -109,14 +109,14 @@ impl CommandContext {
             period_tx_count += block.block().body.kernels().len() - 1;
             period_block_count += 1;
             let st = if prev_block.header().timestamp.as_u64() >= block.header().timestamp.as_u64() {
-                1.0
+                1
             } else {
-                (block.header().timestamp.as_u64() - prev_block.header().timestamp.as_u64()) as f64
+                block.header().timestamp.as_u64() - prev_block.header().timestamp.as_u64()
             };
             let diff = block.accumulated_data.target_difficulty.as_u64();
             period_difficulty += diff;
-            period_solvetime += st as u64;
-            period_hash += diff as f64 / st / 1_000_000.0;
+            period_solvetime += st;
+            period_hash += diff as f64 / st as f64 / 1_000_000.0;
```

Comment:
This can still cause issues since `f64` is more restricted than `u64`, right?

Reply:
if the u64 is really big, then yes converting to f64 might introduce some inaccuracies, but not going to be completely wrong. But here it should still kinda give a near enough answer. 
See this example of the lossy translation to f64: https://stackoverflow.com/questions/75518263/convert-u64-to-f64